packages/block-library/src/categories/edit.js

```diff
@@ -52,7 +52,8 @@ export default function CategoriesEdit( {
 
 	const { records: allTaxonomies, isResolvingTaxonomies } = useEntityRecords(
 		'root',
-		'taxonomy'
+		'taxonomy',
+		{ per_page: 100 }
```

Comment:
```suggestion
		{ per_page: -1 }
```

We should just specify an unbounded query. The taxonomies endpoint doesn't support pagination, and it will return all the records, no matter what `per_page` value is specified.

Unfortunately, there's an old and odd bug in the data package and `per_page: -1` is a workaround for now. Ref: #15413.

P.S. I'm working on this fix, but in the meantime, we could ship this solution.